src/partisan_hiscamp_membership_strategy.erl

```diff
@@ -61,14 +61,23 @@ join(#hiScamp{membership=Membership0, level1=L1, level2=L2}=State0, Node, _NodeS
     MembershipList = membership_list(State),
     {ok, Membership, OutgoingMessages, State}.
 
-distance(#hiScamp{membership=Membership0, heap=Heap, level1=L1, level2=L2}=_State0, Node) ->
-    % A node j joins the system by sending a subscription request to the node s which is closest to it
-    Threshold = 3,
+try_distance(#hiScamp{membership=Membership0, distance=Dist}=_State0, Node) ->
     Dict = get(distance_metrics),
     try dict:find(Node, Dict) of
         {ok, Dist} -> Dist
     catch
          error:E -> {"Error computing distance"}
```

Comment:
You shouldn't be returning strings as errors, you should be using atoms because pattern matching against the string is not going to be readable.  

You should be using the Erlang idiom of using ```{ok, Value}``` vs ```{error, Error}``` style tuples for matching values vs. errors.

Reply:
Ok, I think I fixed it? 
But Im still getting these:
src/partisan_hiscamp_membership_strategy.erl:21: variable 'Dist' is unbound
src/partisan_hiscamp_membership_strategy.erl:77: function try_distance/0 undefined

21     State = #hiScamp{membership=Membership, actor=Identity, heap=Heap, level1 = L1, level2 = L2, distance=Dist},

 77     try try_distance() of

try_distance shouldn't have to take in any parameters, right?

Reply:
try_distance is using both Node and Dict on line 68, so it needs to take them as arguments.

Reply:
ok, I've done that, but its still saying 'function try_distance/2 undefined'

Reply:
It's undefined because of the error on L21.